### PR TITLE
Feature/5 dockerize

### LINKED ISSUE #5: [FEATURE] Dockerize
## Feature Description

Dockerize rust api-gateway

## Definition of Done

The api-gateway is dockerized

### PR DESCRIPTION
**Pull request by TunixR**
This pull request includes significant changes to migrate the project from a Node.js-based setup to a Rust-based setup. The most important changes include modifications to the Docker setup, updates to the project dependencies and build commands, and adjustments to the configuration files to align with the new Rust environment.

### Migration to Rust:

* [`Dockerfile`](diffhunk://#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557L1-R9): Replaced the Node.js base image and setup with a Rust-based build process, including a multi-stage Docker build for the Rust project.
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L10-R74): Updated the installation, development, and production instructions to use Rust commands (`cargo`) instead of Node.js commands (`npm`).

### Configuration Updates:

* [`.github/workflows/docker-deploy.yml`](diffhunk://#diff-ca5936e99aab231027b9d911c6120332b39f993322409f5f6a09929934f7ece9L16-R17): Updated the Docker image names to `api-gateway` to reflect the new project name.
* [`config.yaml.example`](diffhunk://#diff-8820e266c757e766e9a345e6aa55bcd237f1795744ed124b49c5c50434dcdbecL2-R2): Renamed and updated the configuration file to use `0.0.0.0` as the API gateway URL.

### Docker and Environment Setup:

* [`docker-compose.yml`](diffhunk://#diff-e45e45baeda1c1e73482975a664062aa56f20c03dd9d64a827aba57775bed0d3L4-R16): Removed the MongoDB service and updated the service name and configuration for the new `api-gateway`.

### Code Adjustments:

* [`src/config/parser.rs`](diffhunk://#diff-fe0533b7fb6689b2c6c973510eb5bd71e2a04f717832b7f0b3b5d66028b82958L37-R37): `load_config` now reads from a preloaded `&str` value representing the contents of `config.yaml` to allow reading the configuration in production.
* [`src/main.rs`](diffhunk://#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR22-R26): Modified the main function to load the configuration from an included string instead of a file.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
 
-RUN npm ci --production && \
-    rm -rf $(npm get cache)
-
-ENTRYPOINT ["npm", "start"]
\ No newline at end of file
+FROM debian:bookworm-slim
+RUN apt-get update && apt-get install openssl libssl-dev libssl3 && rm -rf /var/lib/apt/lists/*
+COPY --from=builder /usr/local/cargo/bin/api-gateway /usr/local/bin/api-gateway
+CMD ["api-gateway"]
```
Comment (alvarobernal2412) [inline Dockerfile]: It is more advisable to use entrypoints instead of CMD for docker.

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,9 @@
-FROM node:lts-alpine
-
-# TODO: Cambiar el nombre de la carpeta por el nombre de tu proyecto
-WORKDIR /template
-
+FROM rust:1.82 as builder
+WORKDIR /usr/src/api-gateway
```
Comment (alvarobernal2412) [inline Dockerfile]: You don't need to use an absolute path like /usr/src/api-gateway for WORKDIR. Using a shorter path, such as /api-gateway, is perfectly valid and simplifies the Dockerfile. Absolute paths can make directory purposes clearer (e.g., indicating source code), but for simpler projects, a concise path works just as well.

**Review by alvarobernal2412 — CHANGES_REQUESTED** [review]
It looks very good, however, there are still a couple of things to be polished up

File: Dockerfile (hunk 2 above)
Comment (TunixR) [reply Dockerfile]: nit. But ok, I'll change it

File: Dockerfile (hunk 1 above)
Comment (TunixR) [reply Dockerfile]: True. I'll fix it

**Review by adrrf — APPROVED** [review]
nice

**Review by alvarobernal2412 — APPROVED** [review]
James Franco Thumbs Up

**Pull request merged**